werkzeug/wrappers.py

```diff
@@ -1443,12 +1484,43 @@ def make_conditional(self, request_or_environ):
             # wsgiref.
             if 'date' not in self.headers:
                 self.headers['Date'] = http_date()
+            if accept_ranges is True:
+                accept_ranges = "bytes"
+            elif accept_ranges is None:
+                accept_ranges = "none"
+            elif not isinstance(accept_ranges, string_types):
+                accept_ranges = None
+            if accept_ranges is not None:
+                self.headers['Accept-Ranges'] = accept_ranges
+                if self._is_range_request_processable(environ) and complete_length is not None:
+                    parsed_range = parse_range_header(self.headers.get('range'))
+                    if parsed_range is not None:
+                        range_tuple = parsed_range.range_for_length(complete_length)
+                        content_range_header = parsed_range.to_content_range_header(complete_length)
+                        if range_tuple is not None and content_range_header is not None:
+                            content_length = range_tuple[1] - range_tuple[0]
+                            # Be sure not to send 206 response
+                            # if requested range is the full content.
+                            if content_length != complete_length:
+                                self.headers['Content-Length'] = content_length
+                                self.content_range = content_range_header
+                                self.status_code = 206
+                                self._wrap_response(range_tuple[0], content_length)
+                        elif not ignore_invalid_range_requests:
+                            from werkzeug.exceptions import RequestedRangeNotSatisfiable
+                            raise RequestedRangeNotSatisfiable(complete_length)
+                    elif not ignore_invalid_range_requests:
+                        from werkzeug.exceptions import RequestedRangeNotSatisfiable
+                        raise RequestedRangeNotSatisfiable(complete_length)
+                    # If range request is not able to complete but does not
+                    # raise an exception, we get back to basic workflow.
             if self.automatically_set_content_length and 'content-length' not in self.headers:
                 length = self.calculate_content_length()
                 if length is not None:
                     self.headers['Content-Length'] = length
-            if not is_resource_modified(environ, self.headers.get('etag'), None,
-                                        self.headers.get('last-modified')):
+            if self.status_code != 206 and \
+                    not is_resource_modified(environ, self.headers.get('etag'),
+                                             None, self.headers.get('last-modified')):
```

Comment:
~~~python
if (
    self.status_code != 206
    and not is_resource_modified(
        environ, self.headers.get('etag'),
        None, self.headers.get('last-modified')
    )
):
~~~